### PR TITLE
E2E test apps for WinForms and WPF BlazorWebView

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
This is part of #30656

This PR adds two new projects:

 * `src/Components/WebView/Platforms/WinForms/testassets/WinFormsTestApp`
 * `src/Components/WebView/Platforms/Wpf/testassets/WpfTestApp`

They both take a dependency on `BasicTestApp`, which contains Blazor's primary E2E test scenarios:

image

As discussed before, I've clicked through each of them manually to get an overview of which functional areas are already working in WebView and which ones still need some more attention. This is now tracked in https://github.com/dotnet/aspnetcore/issues/31019. Note that actually making all of these work perfectly is out of scope for preview 3 - we just need to know where we are and what should be expected to work in preview 3.

### What about actually wiring up the Selenium tests to this?

The contents of `BasicTestApp` are designed for two consumers: (1) humans who click in the UI manually, and (2) several hundred Selenium tests in the `Microsoft.AspNetCore.Components.E2ETests` project. This PR only takes care of (1), since there isn't currently any way of running the existing Selenium test suite against these WebView-based test apps.

I *did* spend quite a lot of time trying to make the existing Selenium tests run against these test apps, and came to the conclusion that it's not a great thing to do:

* It involves duplicating a *lot* of very intricate and delicate (a.k.a. hacky) infrastructure to take a dependency on Selenium and `selenium-standalone`. Much of it needs to be overlaid with a further layer of hacks to make it run in an entirely new context without clashing with itself since there's also got to be a project reference to `E2ETests` which also imports `E2ETesting.targets` etc.
* It involves either widespread refactoring of the .NET Selenium infrastructure to recognize an entirely different app startup and hosting mechanism, or (what I tried) shared-source-based reuse with compile-time tricks to make it behave differently in certain ways on WebView
* Simply having a project reference to `E2ETests` is extremely unpleasant, because that project transitively references all code that has ever been written in all human history, so its builds are extremely time-consuming and don't support incrementalism. Referencing it brings the same weight.
* We'd have to make per-test-case judgements about what things should run in WebView, and in all future work on `E2ETests`, also account for WebView and try running the test cases there too (perhaps on all 3 desktop platforms, which would quadruple the time to execute E2E tests)
* In the end it still didn't work. After fixing everything that seemed to be in our code, I was getting an error something like "An object is not in the expected state" from deep inside the CoreWebView2 internals. Perhaps with more investigation this could be solved. Maybe it's some kind of timing issue to do with startup.

You can find my efforts to make this work in [this branch](https://github.com/dotnet/aspnetcore/tree/stevesa/webview-enable-existing-e2e-tests), but like I said, it's probably not worth trying to complete this.

What I think we should do instead is what we discussed before, which is setting up a WebView-specific equivalent to `E2ETests` which *only* contains smoke tests for hosting-specific features. We should be validating that (say) the Dispatcher works correctly in all WebView cases, since it's different for each of them, but we shouldn't be validating that (say) cascading parameters work correctly, as there's nothing WebView-specific about that.

### CODE REVIEW

File: src/Components/WebView/Platforms/WindowsForms/testassets/WinFormsTestApp/Form1.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Windows.Forms;
+using System.Net.Http;
+using Microsoft.AspNetCore.Components.WebView.WindowsForms;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace WinFormsTestApp
+{
+    public partial class Form1 : Form
```
Comment (javiercn) [inline src/Components/WebView/Platforms/WindowsForms/testassets/WinFormsTestApp/Form1.cs:11]: Give it a proper name, otherwise it's like `Class1` and people will ignore it or delete it.

**Review by javiercn — APPROVED** [review]
One a minor comment, but looks great!

WRT to the testing piece, how hard would you think would be to setup a CTI test for it?

Based on the list on the issue seems we are in "good shape" here?

File: src/Components/WebView/Platforms/WindowsForms/testassets/WinFormsTestApp/Form1.cs (hunk 1 above)
Comment (SteveSandersonMS) [reply src/Components/WebView/Platforms/WindowsForms/testassets/WinFormsTestApp/Form1.cs:11]: I was following on from the conventions in other projects like in `BlazorWinFormsApp`. Doesn't seem problematic right now but we could certainly change them all at some point.

**Comment by SteveSandersonMS** [comment]
> WRT to the testing piece, how hard would you think would be to setup a CTI test for it?

It would take at least a day of work to write up all the steps, maybe two. It took me about 2 hours just to click through it all pretty quickly on a one-time basis. We would also want to tweak lots of the test cases in `BasicTestApp` to provide more human-friendly UIs.

>  Based on the list on the issue seems we are in "good shape" here?

Once we have static web assets support, I'd say we're in adequate shape for preview 3. There will be some known edge case problems like the dispatcher thing, and we might still want to fix the InputFile issue since it might be trivial.

**Comment by SteveSandersonMS** [comment]
@Eilon Tagging you just FYI. These new test apps may be useful to you.

**Comment by Eilon** [comment]
This is great!!!
